OMF_API/Program.cs

```diff
@@ -1,601 +1,268 @@
+                        string omfContainerString = $"[{JsonConvert.SerializeObject(omfContainer)}]";
+                        sendMessageToOmfEndpoint(endpoint, "container", omfContainerString);
+                    }
+                }
 
+                // Step 7 - Send OMF Data
                 int count = 0;
-                string value = "";
-                while (count == 0 || ((!test) && count < 2))
+                //send data to all endpoints forever if this is not a test
+                while (!test || count < 2)
                 {
-                    //step 9 
-                    var val = create_data_values_for_first_dynamic_type("Container1");
-                    value = val;
+                    /*
+                    * This is where custom loop logic should go.
+                    * The getData call should also be customized to populate omfData with relevant data.
+                    * */
 
-                    sendValue("data", val);
-                    sendValue("data", create_data_values_for_first_dynamic_type("Container2"));
-                    sendValue("data", create_data_values_for_second_dynamic_type("Container3"));
-                    sendValue("data", create_data_values_for_third_dynamic_type("Container4"));
-                    if (sendingToOCS)
-                        sendValue("data", create_data_values_for_NonTimeStampIndexAndMultiIndex_type("Container5", "Container6"));
-                    Thread.Sleep(1000);
-                    count += 1;
-                }
-                CheckValues(value);
+                    foreach (var omfDatum in omfData)
+                    {
+                        // retrieve data
+                        getData(omfDatum);
+
+                        foreach (var endpoint in endpoints)
+                        {
+                            // send data
+                            string omfDatumString = $"[{JsonConvert.SerializeObject(omfDatum)}]";
+                            sendMessageToOmfEndpoint(endpoint, "data", omfDatumString);
+                        }
 
-            }
-            catch (Exception ex)
-            {
-                Console.WriteLine(ex.ToString());
+                        // record the values sent if this is a test
+                        if (test && count == 1)
+                            lastSentValues.Add((string)omfDatum.containerid, omfDatum);
 
-                toThrow = ex;
-                success = false;
-            }
-            finally
-            {
-                Console.WriteLine("Deleting");
-                //step 10
-                try
-                {
-                    deleteTypesAndContainers();
-                }
-                catch (Exception ex)
-                {
-                    Console.WriteLine(ex.ToString());
+                    }
 
-                    toThrow = ex;
-                    success = false;
+                    count++;
+                    Thread.Sleep(1000);
```

Comment:
Why do we wait a second between each value send?
If this is intended, consider moving the `1000` into a constant, like `int SEND_SLEEP = 1000;`

Reply:
This is an arbitrary pause. I will go with a constant to make it configurable